solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -519,7 +525,11 @@ private Request makeRequest(SolrRequest<?> solrRequest, String collection)
 
   private void decorateRequest(Request req, SolrRequest<?> solrRequest) {
     req.header(HttpHeader.ACCEPT_ENCODING, null);
-    req.timeout(idleTimeout, TimeUnit.MILLISECONDS);
+    if (requestTimeout > 0) {
```

Comment:
If someone accidentally sets this to 0, we'd just be ignoring it?

Reply:
I don't think it makes much difference. 0 means timeout disabled, but our client will only wait for `idleTimeout` (see `listener.get(idleTimeout, TimeUnit.MILLISECONDS);`), so we'll timeout anyway, and at that point, probably better to timeout the request, since the client won't consume it anyway. WDYT?